m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,773 @@
+
+PROCEDURE CLEARTRANVARS
+[* Clear the variables storing transactional detail
+*]
+ FOR TMPLOOP=1 TO TCCOUNT
+  DO
+   TCCOMMENT(TMPLOOP)=""
+   TCSTMTDESCR(TMPLOOP)=""
+   TCIDTYPE(TMPLOOP)=0
+   TCID(TMPLOOP)=""
+  END
+ TCCOUNT=0
+ TOACCOUNT=""
+ TOSLIDTYPE=""
+ TOSLID=""
+ TOTRANAMOUNT=$0.00
+END [PROCEDURE]
+
+PROCEDURE GETSHARETRANSFERRECS
+[* Read through FM History for Share transfer records created within the
+** desired date range and by the desired user number. At each change in
+** the sequence number, evaluate the current data saved. If a targeted
+** transaction, then record, clear the record arrays and then read through
+** the next record until the FM History post date < target start date.
+*]
+ CALL CLEARFIELDS
+ LASTSEQ=0
+
+ FOR EACH FMHISTORY WITH (FMHISTORY:POSTDATE>=TARGETSTARTDATE AND
+                          FMHISTORY:POSTDATE<=TARGETENDDATE AND
+                          FMHISTORY:FMTYPE=0 AND [CREATION]
+                          FMHISTORY:RECORDTYPE=8 AND [SHARE TRANSFER RECORD]
+                          FMHISTORY:USERNUMBER=TARGETSYMXUSER)
+  DO
+   IF FMHISTORY:SEQUENCENUMBER<>LASTSEQ THEN
+    DO
+     IF LASTSEQ>0 AND
+        FMHXFERTYPE=3 THEN
+      CALL PROCESSRECORD
+     CALL CLEARFIELDS
+
+     LASTSEQ=FMHISTORY:SEQUENCENUMBER
+    END
+
+   IF CHARACTERSEARCH(UPPERCASE(FMHISTORY:FIELDNAME),"NEW LOCATOR")>0 THEN
+    DO
+     FMHNEWLOC=FMHISTORY:NEWNUMBER
+     FMHFROMACCOUNT=FMHISTORY:ACCOUNTNUMBER
+     FMHFROMIDTYPE=FMHISTORY:IDTYPE
+     FMHFROMID=FMHISTORY:ID
+     FMHPOSTDATE=FMHISTORY:POSTDATE
+    END
+   ELSE IF FMHISTORY:FIELDNUMBER=14 THEN  [Locator]
+    FMHNEWLOC=FMHISTORY:NEWNUMBER
+   ELSE IF FMHISTORY:FIELDNUMBER=13 AND   [Day 2 - can be multiple entries]
+           FMHISTORY:OLDNUMBER=0 THEN
+    FMHDAY2=FMHISTORY:NEWNUMBER
+   ELSE IF FMHISTORY:FIELDNUMBER=12 AND   [Day 1 - can be multiple entries]
+           FMHISTORY:OLDNUMBER=0 THEN
+    FMHDAY1=FMHISTORY:NEWNUMBER
+   ELSE IF FMHISTORY:FIELDNUMBER=11 THEN  [Xfer Freq]
+    FMHXFERFREQ=FMHISTORY:NEWNUMBER
+   ELSE IF FMHISTORY:FIELDNUMBER=10 THEN  [Next Date]
+    FMHNEXTXFERDATE=FMHISTORY:NEWDATE
+   ELSE IF FMHISTORY:FIELDNUMBER=08 THEN  [Expiration Date]
+    FMHEXPIREDATE=FMHISTORY:NEWDATE
+   ELSE IF FMHISTORY:FIELDNUMBER=07 THEN  [Effective Date]
+    FMHEFFECTIVEDATE=FMHISTORY:NEWDATE
+   ELSE IF FMHISTORY:FIELDNUMBER=05 THEN  [Amount]
+    FMHXFERAMOUNT=FMHISTORY:NEWMONEY
+   ELSE IF FMHISTORY:FIELDNUMBER=04 THEN  [ID]
+    FMHTOID=FMHISTORY:NEWCHARACTER
+   ELSE IF FMHISTORY:FIELDNUMBER=03 THEN  [ID Type]
+    FMHTOIDTYPE=FMHISTORY:NEWNUMBER
+   ELSE IF FMHISTORY:FIELDNUMBER=02 THEN  [Account]
+    FMHTOACCOUNT=FMHISTORY:NEWCHARACTER
+   ELSE IF FMHISTORY:FIELDNUMBER=01 AND   [Transfer Type - can be multiple entries]
+           FMHISTORY:OLDNUMBER=0 THEN
+    FMHXFERTYPE=FMHISTORY:NEWNUMBER
+  END
+ UNTIL FMHISTORY:POSTDATE<TARGETSTARTDATE [FOR EACH FMHISTORY]
+
+ IF FMHXFERTYPE=3 THEN [Catch last transaction - if type 3]
```

Comment:
@rrobison44 I don't believe so. At the top of the FOR EACH FMHISTORY loop the need to check for the last sequence number > 0 is because at the start of the loop, the LASTSEQ will always be 0 and at that point there will be no values in the variables to evaluate and print out. At the end of the FOR EACH FMHISTORY loop, we want to force the last stored record to print out. If there was only one record found in the FM History search, the LASTSEQ would be 0 and then the record wouldn't get printed out.